spring-cloud-gcp-docs/src/main/asciidoc/pubsub.adoc

```diff
@@ -26,13 +26,12 @@ dependencies {
 
 === Pub/Sub operations abstraction
 
-`PubSubTemplate` is an abstraction that allows Spring users to use Google Cloud Pub/Sub without
+`PubSubOperations` is an abstraction that allows Spring users to use Google Cloud Pub/Sub without
```

Comment:
I don't mind actually calling it `PubSubTemplate`, even though interface is `PubSubOperations`. I think "template" is a more familiar term for Spring users.

Reply:
could say "PubSubTemplate implements PubSubOperations, which is an abstraction that allows..."

Reply:
One thing to have in mind is that the Spring Boot GCP Core Starter injects a `PubSubOperations` instance, not a `PubSubTemplate`, which is why I decided to make `PubSubOperations` more widespread.
We can change the Core Starter to redundantly provide a `PubSubTemplate` bean, which is an exact copy of the `PubSubOperations` it currently provides, to mitigate this risk.
WDYT?

Reply:
`PubSubTemplate` gets auto-wired just fine as is right now. So, I would take Mark's suggestion here.

Reply:
It actually doesn't, which is why I'm suggesting we make the Core Starter provide it.

Reply:
So as we found out last week, auto-wiring a `PubSubTemplate` without providing a `@Bean` for it has different behaviour depending on if a class is `@Configuration` or not.

If you keep reading the paragraph, for the remainder of the section, the main subject is `PubSubTemplate`, not `PubSubOperations`. Is it not good enough?